File: organized_code/STM Paper 2017/Matlab/Data Visualization/Factor_Deficiency.m

% Determine if trauma patient factor levels are
% a) less than, within, or greater than control plasma factor levels +/- 10%
% b) less than, within, or greater than 20 normals factor levels +/- 10% 

clear all;
close all;
clc;

% Control Plasma Percentage
Control(1) = 87; %II
Control(2) = 80; %V
Control(3) = 87; %VII
Control(4) = 92; %VIII
Control(5) = 91; %IX
Control(6) = 90; %X
Control(7) = 88; %ATIII

% Read in trauma concentrations
MCADataValues = xlsread('../../Data/Processed/CAT_Trauma','Fits','G2:M41');

% Create an array, rows are trauma patient, cols are factors
% -1 if below range, 0 if within range, +1 if above range

% Initialize
TraumaPatientWithinControlRange = zeros(40,7);

for TraumaPatientNum = 1:40
    for FactorNum = 1:7
        if MCADataValues(TraumaPatientNum,FactorNum) < 0.9*Control(FactorNum)
            TraumaPatientWithinControlRange(TraumaPatientNum,FactorNum) = -1;
        elseif MCADataValues(TraumaPatientNum,FactorNum) > 1.1*Control(FactorNum)
            TraumaPatientWithinControlRange(TraumaPatientNum,FactorNum) = 1;
        end
    end
end

TraumaPatientWithinControlRangeMin = min(TraumaPatientWithinControlRange,[],2)
%37/40 have a factor deficiency this way

TraumaPatientWithinControlRangeMax = max(TraumaPatientWithinControlRange,[],2)
%31/40 have a factor excess this way

% Read in normal concentrations
NormalConc = xlsread('../../Data/Processed/CAT_Normals','Fits','G2:M21');

MeanNormalConc = mean(NormalConc);
StdDevNormalConc = std(NormalConc);

% Create an array, rows are trauma patient, cols are factors
% -1 if below range, 0 if within range, +1 if above range

% Initialize
TraumaPatientWithinNormalsRange = zeros(40,7);

for TraumaPatientNum = 1:40
    for FactorNum = 1:7
        if MCADataValues(TraumaPatientNum,FactorNum) < (MeanNormalConc(FactorNum)-StdDevNormalConc(FactorNum))
            TraumaPatientWithinNormalsRange(TraumaPatientNum,FactorNum) = -1;
        elseif MCADataValues(TraumaPatientNum,FactorNum) > (MeanNormalConc(FactorNum)+StdDevNormalConc(FactorNum))
            TraumaPatientWithinNormalsRange(TraumaPatientNum,FactorNum) = 1;
        end
    end
end

TraumaPatientWithinNormalsRangeMin = min(TraumaPatientWithinNormalsRange,[],2)
%32/40 have a factor deficiency this way

TraumaPatientWithinNormalsRangeMax = max(TraumaPatientWithinNormalsRange,[],2)
%39/40 have a factor excess this way

File: organized_code/STM Paper 2017/Python/Data Visualization/Factor_Deficiency.py
# Determine if trauma patient factor levels are
# a) less than, within, or greater than control plasma factor levels +/- 10%
# b) less than, within, or greater than 20 normals factor levels +/- 10%

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

'''
Control Plasma Percentage
Control - numpy.array:
          Control[0] -> Factor II
          Control[1] -> Factor V
          Control[2] -> Factor VII
          Control[3] -> Factor VIII
          Control[4] -> Factor IX
          Control[5] -> Factor X
          Control[6] -> Factor ATIII
'''
Control = np.zeros(7)
Control[0] = 87
Control[1] = 80
Control[2] = 87
Control[3] = 92
Control[4] = 91
Control[5] = 90
Control[6] = 88

# Read in trauma concentrations
CATN = '../../Data/Processed/CAT_Normals.xlsx'
CATT = '../../Data/Processed/CAT_Trauma.xlsx'
dfN = pd.read_excel(CATN, sheet_name='Fits', usecols="G:M").dropna()
dfT = pd.read_excel(CATT, sheet_name='Fits', usecols="G:M").dropna()
MCADataValues = dfT[:42].to_numpy()

# Create an array, rows are trauma patient, cols are factors
# -1 if below range, 0 if within range, +1 if above range

# Initialize
TraumaPatientWithinControlRange = np.zeros((40,7));
for TraumaPatientNum in range(40):
    for FactorNum in range(7):
        if MCADataValues[TraumaPatientNum, FactorNum] < 0.9 * Control[FactorNum]:
            TraumaPatientWithinControlRange[TraumaPatientNum, FactorNum] = -1;
        elif MCADataValues[TraumaPatientNum, FactorNum] > 1.1 * Control[FactorNum]:
            TraumaPatientWithinControlRange[TraumaPatientNum, FactorNum] = 1;

TraumaPatientWithinControlRangeMin = np.min(TraumaPatientWithinControlRange,axis=1)
TraumaPatientWithinControlRangeMax = np.max(TraumaPatientWithinControlRange,axis=1)
# print(TraumaPatientWithinControlRangeMax)

NormalConc = dfN[:42].to_numpy()
MeanNormalConc = np.mean(NormalConc, axis=0)
StdDevNormalConc = np.std(NormalConc, axis=0, ddof=1)
# Create an array, rows are trauma patient, cols are factors
# -1 if below range, 0 if within range, +1 if above range

# Initialize
TraumaPatientWithinNormalsRange = np.zeros((40,7));
TraumaPatientNum = 0
FactorNum = 0
for TraumaPatientNum in range(40):
    for FactorNum in range(7):
        if MCADataValues[TraumaPatientNum, FactorNum] < (MeanNormalConc[FactorNum] - StdDevNormalConc[FactorNum]):
            TraumaPatientWithinNormalsRange[TraumaPatientNum, FactorNum] = -1;
        elif MCADataValues[TraumaPatientNum, FactorNum] > (MeanNormalConc[FactorNum] + StdDevNormalConc[FactorNum]):
            TraumaPatientWithinNormalsRange[TraumaPatientNum, FactorNum] = 1;

TraumaPatientWithinNormalsRangeMin = np.min(TraumaPatientWithinNormalsRange,axis=1)
TraumaPatientWithinNormalsRangeMax = np.max(TraumaPatientWithinNormalsRange,axis=1)
# print(StdDevNormalConc)
print(TraumaPatientWithinNormalsRangeMin)
